src/hooks/useSetFocus/index.js

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useEffect, useState } from "react";
+import useEventListener from "../useEventListener";
+import { usePrevious } from "hooks";
+
+export default function useSetFocus({ ref, focusCallback, blurCallback }) {
+  const [isFocused, setIsFocused] = useState(false);
+  const isFocusedPrev = usePrevious(isFocused);
+
+  useEffect(() => {
+    if (isFocusedPrev === undefined) {
```

Comment:
I remember we put the logic inside of use effect (calling callback functions) instead of on the onFocus and onBlur function for a reason (to be sure the value of isFocused is also changed).
let's write a comment about this because it is not obvious